src/camera.rs

```diff
@@ -47,6 +47,14 @@ impl Default for FpsCamera {
     }
 }
 
+fn flatten_xy(mut dir: Vec3, default: Vec3) -> Vec3 {
+    dir[2] = 0.0;
+    match dir.try_normalize() {
+        Some(v) => v,
+        None => default,
+    }
```

Comment:
`dir.try_normalize().unwrap_or(default)` is more idiomatic here

Reply:
Nice. I knew there had to be something.